src/NUnitFramework/framework/Internal/Tests/TestSuite.cs

```diff
@@ -117,7 +117,9 @@ public TestSuite(TestSuite suite, ITestFilter filter)
                 {
                     if(child.IsSuite)
                     {
-                        TestSuite childSuite = new TestSuite(child as TestSuite, filter);
+                        TestSuite childSuite = child.GetType() == typeof(TestFixture)
+                            ? new TestFixture(child as TestSuite, filter)
+                            : new TestSuite(child as TestSuite, filter);
```

Comment:
These should be direct casts. Direct casts fail closer to the source of the problem and are information-rich. Safe casts are only useful if `null` is a valid value; otherwise, they produce a much harder to understand null reference exception at a later point in time.

Reply:
Agreed. With the new copy method, these casts go away anyway.